### DISCUSSION TITLE
Charge Voltage Limit

### DISCUSSION THREAD

**Discussion by hansgeorgk** [Louisvdw/dbus-serialbattery#415] [Q&A] [answered]
Hi,

I updated to v0.14.3 and I realized that CVL in the Remote console is set using this Value:
FLOAT_CELL_VOLTAGE = 3.38
<img width="513" alt="Bildschirm­foto 2023-01-14 um 09 40 30" src="https://user-images.githubusercontent.com/100286224/212477022-7358c0f4-3222-43ed-9ea9-3e3c90acde6a.png">

I have set
~~~
# Charge current control management referring to cell-voltage enable (True/False).
CCCM_CV_ENABLE = True
# Discharge current control management referring to cell-voltage enable (True/False).
DCCM_CV_ENABLE = True
~~~

all other charge control settings are set to False.

is this the intended behavior ?
cheers
HG

**Comment by MisterX1000** [comment]
Hey HG,

same behavior for my 16 cell installation here. After installation of v.0.14.3 the "Float" voltage is used as CVL. With v.0.14.2 the CVL is like expected!

MIN_CELL_VOLTAGE = 2.9
_MAX_CELL_VOLTAGE = 3.45_
**FLOAT_CELL_VOLTAGE = 3.35**
MAX_VOLTAGE_TIME_SEC = 15*60
SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT = 90

![image](https://user-images.githubusercontent.com/112106667/212568743-22330692-d5b2-45c3-a393-1747f3d06422.png)

Here you can see that I installed the v.0.14.3 at 12.Jan - there was the CVL "jump" down.

![image](https://user-images.githubusercontent.com/112106667/212569008-34abbe04-637d-47c1-968b-bacee9abd9bb.png)

**Comment by MisterX1000** [comment]
I just post this problem also in the "Issues" section. If this is not wanted, just delete my second post.

Regards
Alex

**Comment by rcrc1978** [comment]
Same problem for me, added to your problem and a few others.

#421     
#317 
#312 
#297

**Comment by Louisvdw** [comment]
The FLOAT_CELL_VOLTAGE is suppose to only apply if CVCM_ENABLE = true, so I'll investigate what is happening.
CCCM_CV_ENABLE should not influence it.

**Comment by redmoon2711** [comment]
Same problem after set "LINEAR_LIMITATION_ENABLE = True"  CVL jump to  53.6V

After set  "LINEAR_LIMITATION_ENABLE = False" and reinstall with 
#>sh /data/etc/dbus-serialbattery/reinstalllocal.sh 

CVL jump to 55.2V (16 * MAX_CELL_VOLTAGE)

**Reply by pkali1** [reply]
I have solved this in battery.py.  Louis please could you update the code for your next release (my is not including "configurable changes from latest code).
Code is here. 
[https://github.com/Louisvdw/dbus-serialbattery/issues/421#issuecomment-1454842090]

**Comment by mr-manuel** [comment, marked as answer]
Hi, please test the fix in https://github.com/Louisvdw/dbus-serialbattery/issues/421
